### PR TITLE
liquidity: fix flaky autoloop test

### PR DESCRIPTION
**Pull request by starius**
This failure became normal recently:

```
=== RUN   TestAutoLoopInEnabled
    autoloop_testcontext_test.go:318:
        	Error Trace:	/home/runner/work/loop/loop/liquidity/autoloop_testcontext_test.go:318
        	            				/home/runner/work/loop/loop/liquidity/autoloop_test.go:804
        	Error:      	Not equal:
        	            	expected: 80000
        	            	actual  : 160000
        	Test:       	TestAutoLoopInEnabled
    autoloop_testcontext_test.go:318:
        	Error Trace:	/home/runner/work/loop/loop/liquidity/autoloop_testcontext_test.go:318
        	            				/home/runner/work/loop/loop/liquidity/autoloop_test.go:804
        	Error:      	Not equal:
        	            	expected: 160000
        	            	actual  : 80000
        	Test:       	TestAutoLoopInEnabled
    autoloop_testcontext_test.go:343:
        	Error Trace:	/home/runner/work/loop/loop/liquidity/autoloop_testcontext_test.go:343
        	            				/home/runner/work/loop/loop/liquidity/autoloop_test.go:804
        	Error:      	Should be true
        	Test:       	TestAutoLoopInEnabled
```

The root cause is them the order of items in c.quoteRequestIn depends on the order of loopInBuilder.buildSwap calls, which depends on the order of channel handling in Manager.SuggestSwaps, which depends on the order of map traversal, which is not determitistic.

Since in the test all the amounts are different, I used amount as a key and put the expected calls into a map using amount as a key. When I extract an item from c.quoteRequestIn channel, I find the corresponding item in the map and remove it. All other logic is preserved.

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: liquidity/autoloop_testcontext_test.go [hunk 1]
```diff
@@ -312,9 +312,27 @@ func (c *autoloopTestCtx) autoloop(step *autoloopStep) {
...
-	// log them.
+	// log them. The order in c.quoteRequestIn is not deterministic,
+	// it depends on the order of map traversal (map peerChannels in
+	// method Manager.SuggestSwaps). So receive from the channel an item
+	// and then find a corresponding expected item, using amount as a key.
+	amt2expected := make(map[btcutil.Amount]quoteInRequestResp)
 	for _, expected := range step.quotesIn {
+		// Make sure all amounts are unique.
+		_, has := amt2expected[expected.request.Amount]
+		require.False(c.t, has)
```
Comment (bhandras) [inline liquidity/autoloop_testcontext_test.go:323]: nit: note that there's also `require.NotContains` which might just do this for us.

**Review by bhandras — APPROVED** [review]
LGTM 💯

**Review by hieblmi — APPROVED** [review]
tACK `go test -v ./liquidity/... -race -run TestAutoLoopInEnabled -count 100`

File: liquidity/autoloop_testcontext_test.go (hunk 1 above)
Comment (starius) [reply liquidity/autoloop_testcontext_test.go]: Fixed

**Pull request merged**